linuxPackages.tsme-test: init at 6.6.68-unstable-2022-12-07

lynatic1337 opened the pull request:
## Things done
Add https://github.com/AMDESE/mem-encryption-tests to nixpkgs. This adds a kernel module that allows to check whether TSME is enabled and working as intended on supported systems while booted.
Note: Testing this requires an supported AMD CPU and supported BIOS. For testing, please load the kernel module and check whether /sys/kernel/tsme exists.
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

lynatic1337 committed:
maintainers: add lyn

lynatic1337 committed:
linuxPackages.tsme-test: init at 6.6.68-unstable-2022-12-07

FliegendeWurst commented on pkgs/os-specific/linux/mem-encryption-tests/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+stdenv.mkDerivation {
+  pname = "mem-encryption-tests";
+  version = "0-master-2022-12-07";
```
```suggestion
  version = "0-unstable-2022-12-07";
```

FliegendeWurst commented on pkgs/os-specific/linux/mem-encryption-tests/default.nix:
```diff
@@ -0,0 +1,36 @@
+    "INSTALL_MOD_PATH=$(out)"
+  ];
+
+  installPhase = ''
+    mkdir -p $out/lib/modules/${kernel.modDirVersion}/extra
+    cp tsme-test.ko $out/lib/modules/${kernel.modDirVersion}/extra
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/lib/modules/${kernel.modDirVersion}/extra
    cp tsme-test.ko $out/lib/modules/${kernel.modDirVersion}/extra

    runHook postInstall
  '';
```

FliegendeWurst commented on pkgs/os-specific/linux/mem-encryption-tests/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  meta = with lib; {
+    description = "Tests related to AMD Memory Encryption";
+    license = licenses.gpl2;
```
```suggestion
    license = licenses.gpl2Only;
```

FliegendeWurst commented on pkgs/os-specific/linux/mem-encryption-tests/default.nix:
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, linuxPackages, kernel ? linuxPackages.kernel }:
```
```suggestion
{
  lib,
  stdenv,
  fetchFromGitHub,
  linuxPackages,
  kernel ? linuxPackages.kernel,
}:
```

FliegendeWurst reviewed (changes requested):
Just some general suggestions. Unfortunately I can't test the module.

lynatic1337 commented:
Thanks for the suggestions and fixes @zi3m5f @FliegendeWurst, I've included them all in this new revision <3
Additionally I fixed the wrong attributes that somehow made it into this file by accident, the revision before would build the module for **some** kernel but not the kernel used in the system configuration.

zi3m5f reviewed (changes requested):
- [x] Checked `githubId`

I think the package name should contain `tsme` to be easier to find.

lynatic1337 commented:
@zi3m5f thanks for the other suggestions, implemented them as well. Package is now renamed to tsme-test, in case AMD decides to add separate modules/tools for other memory-encryption systems we have to change the name back but considering the fact they haven't added any other modules in years...

INSTALL_MOD_PATH was left over from an old version and does nothing, apologies for not noticing.

zi3m5f reviewed (approved):
Thanks, LGTM :)

Build with latest kernel and checked with `modinfo`:
> filename:       result/lib/modules/6.11.5/extra/tsme-test.ko
description:    TSME test module
license:        GPL
author:         Advanced Micro Devices, Inc.
depends:        
retpoline:      Y
name:           tsme_test
vermagic:       6.11.5 SMP preempt mod_unload 
parm:           debug: print extra debug information - any non-zero value (uint)
